lib/publiccloud/azure.pm

```diff
@@ -69,12 +69,11 @@ sub vault_create_credentials {
     my ($self) = @_;
 
     record_info('INFO', 'Get credentials from VAULT server.');
-    my $res = $self->vault_api('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/azure/creds/openqa-role', method => 'get');
-    $self->vault_lease_id($res->{lease_id});
-    $self->key_id($res->{data}->{client_id});
-    $self->key_secret($res->{data}->{client_secret});
+    my $data = $self->vault_get_secrets('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/azure/creds/openqa-role');
+    $self->key_id($data->{client_id});
+    $self->key_secret($data->{client_secret});
 
-    $res = $self->vault_api('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/secret/azure/openqa-role', method => 'get');
+    my $res = $self->vault_api('/v1/' . get_var('PUBLIC_CLOUD_VAULT_NAMESPACE', '') . '/secret/azure/openqa-role', method => 'get');
```

Comment:
I know it is azure :) but why we doing same thing second time ?

Reply:
It is not the same.
1) '/azure/creds/openqa-role'
2) '/secret/azure/openqa-role'

We need the tenant and subscription id, but they do not come with the credentials. So we just store them in a kv `secret-engine`.

Reply:
ok I see